lib/capabilities.js

```diff
@@ -0,0 +1,53 @@
+/**
+* Class for checking capabilities in a capability statement
+*
+*/
+class Capabilities {
+  /**
+  * Access capabilities
+  *
+  * @param {Object} capabilityStatement - capability statement FHIR resource
+  */
+  constructor(capabilityStatement) {
+    this.capabilityStatement = capabilityStatement;
+  }
+
+  /**
+  * Return capability support information based on the capability statement.
+  *
+  * @param {Object} params - The capability parameters.
+  * @param {String} params.resourceType - Resource type (e.g. 'Patient', 'Observation').
+  * @param {Object} params.capabilityType - Capability type (e.g. 'interaction', 'searchParam').
+  * @param {Object} params.where - Parameters to check additional level of compatibility, optional.
+  * @param {String} params.where.code - Specify a code (e.g., 'read' for interaction), optional.
+  * @param {String} params.where.name - Specify a name (e.g., 'address' for searchParam), optional.
+  *
+  * @return {Boolean|Array} Determination of 'where'-specified capability OR list of capabilities.
```

Comment:
Returning a boolean or an array makes me a bit uncomfortable. I'd prefer to split up the two sets of functionality. Perhaps three functions:

- something like `supports` which takes a resource type, interaction, and optional search param and returns a boolean.
- something like `interactions` which takes a resource type and returns an array of supported interactions
- something like `searchParams` which takes a resource type and returns an array of supported search params